Adding tests of cryptsetup

nasastry opened the pull request:
security/cryptsetup-tests.py adds cryptsetup tests

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

nasastry commented:
Fail scenario:
```
JOB ID     : 3cd0d0e13a531c7bb73c2753b95c44120916ac7e
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T05.42-3cd0d0e/job.log
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:292: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp.
  FutureWarning)
 (1/1) cryptsetup-tests.py:CryptSetup.test:  FAIL: 9 test(s) failed, please refer to the log (597.23 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T05.42-3cd0d0e/results.html
JOB TIME   : 597.28 s
```

Pass scenario:
```
JOB ID     : 2c41787c6e852f7e2ca832f8a94bab3f9eff551d
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T04.57-2c41787/job.log
 (1/1) cryptsetup-tests.py:CryptSetup.test:  PASS (3132.31 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T04.57-2c41787/results.html
JOB TIME   : 3133.27 s
```

harish-24 commented on security/cryptsetup-tests.py:
```diff
@@ -0,0 +1,72 @@
+        '''
+        Install the basic packages to support cryptsetup
+        '''
+
```
empty line not needed

harish-24 commented on security/cryptsetup-tests.py:
```diff
@@ -0,0 +1,72 @@
+                         "device-mapper-libs", "popt-devel", "json-c",
+                         "json-c-devel", "libblkid-devel"])
+        else:
+            self.cancel("Install the package for cryptsetup supported\
```
Suggestion: From this test-case perspective "Unsupported distro" would be a better message here?

harish-24 commented on security/cryptsetup-tests.py:
```diff
@@ -0,0 +1,72 @@
+                self.cancel('%s is needed for the test to be run' % package)
+
+        url = "https://gitlab.com/cryptsetup/cryptsetup/"
+
```
empty line unnecessary again

nasastry commented:
@harish-24 comments taken care. Please have a look. Thanks!!

```
JOB ID     : 4dc4945fe7f967611e6c54b0f3c4bbd63e2de89d
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-23T03.52-4dc4945/job.log
 (1/1) cryptsetup-tests.py:CryptSetup.test:  FAIL: 9 test(s) failed, please refer to the log (600.99 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-23T03.52-4dc4945/results.html
JOB TIME   : 601.12 s
```

PraveenPenguin commented on security/cryptsetup-tests.py:
```diff
@@ -0,0 +1,69 @@
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = "https://gitlab.com/cryptsetup/cryptsetup/"
+        git.get_repo(url, destination_dir=self.logdir)
```
just one suggestion we should not pollute logdir

nasastry replied:
@PraveenPenguin instead of logdir, what are the other dirs can be used? Thanks!!

PraveenPenguin replied:
`self.workdir`  is ideal to use for storing download file

nasastry commented:
Job log after changes:
```
JOB ID     : 975865d5074ff61a1aa2c206b0291b08b7819d6e
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T06.42-975865d/job.log
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:292: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp.
  FutureWarning)
 (1/1) cryptsetup-tests.py:CryptSetup.test:  FAIL: 9 test(s) failed, please refer to the log (599.54 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T06.42-975865d/results.html
JOB TIME   : 599.60 s
```

nasastry commented:
@PraveenPenguin logdir changed to workdir ... please have a look, Thanks!!
@harish-24 Please go through the changes requested by you. Thanks!!

PraveenPenguin reviewed (approved):
LGTM

nasastry commented:
```
JOB ID     : e442ae6218e8212ec5faa36329e16b80081d95e6
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T06.18-e442ae6/job.log
 (1/1) cryptsetup-tests.py:CryptSetup.test:  PASS (3185.42 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T06.18-e442ae6/results.html
JOB TIME   : 3186.24 s
```

harish-24 commented on security/cryptsetup-tests.py:
```diff
@@ -0,0 +1,69 @@
+            if 'FAIL:' in line:
+                count += 1
+                self.log.info(line)
+        if count > 0:
```
Missed this earlier, `If count:`.  Otherwise looks good.

nasastry commented:
@harish-24 addressed the above comment.

```
JOB ID     : 31911fe68faf1c3d4f24a26d24f18a7b350e6dc6
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-28T02.24-31911fe/job.log
 (1/1) cryptsetup-tests.py:CryptSetup.test: FAIL: 9 test(s) failed, please refer to the log (595.92 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-28T02.24-31911fe/results.html
JOB TIME   : 596.01 s
```
